src/extractor/guidance/sliproad_handler.cpp

```diff
@@ -29,273 +32,720 @@ SliproadHandler::SliproadHandler(const IntersectionGenerator &intersection_gener
+    const auto heigth = haversineDistance(second, third);
+
+    const auto area = (length * heigth) / 2.;
+
+    // Everything below is data issue - there are some weird situations where
+    // nodes are really close to each other and / or tagging ist just plain off.
+    const constexpr auto MIN_SLIPROAD_AREA = 3.;
+
+    if (area < MIN_SLIPROAD_AREA || area > max_area)
+        return false;
+
+    return true;
+}
+
+bool SliproadHandler::isValidSliproadLink(const ConnectedRoad &sliproad,
+                                          const ConnectedRoad &first,
+                                          const ConnectedRoad &second) const
+{
+    // If the Sliproad is not a link we don't care
+    const auto sliproad_data = node_based_graph.GetEdgeData(sliproad.eid);
+    if (!sliproad_data.road_classification.IsLinkClass())
+        return true;
+
+    // otherwise neither the first road leading to the intersection we shortcut
+    const auto first_road_data = node_based_graph.GetEdgeData(first.eid);
+    if (first_road_data.road_classification.IsLinkClass())
+        return false;
+
+    // not the second road coming from the intersection we shotcut must be links
+    const auto second_road_data = node_based_graph.GetEdgeData(second.eid);
+    if (second_road_data.road_classification.IsLinkClass())
+        return false;
+
+    return true;
+}
+
+bool SliproadHandler::canBeTargetOfSliproad(const Intersection &intersection)
+{
+    // Example to handle:
+    //       .
+    // a . . b .
+    //  `    .
+    //    `  .
+    //       c    < intersection
+    //       .
+    //
+
+    // One outgoing two incoming
+    if (intersection.size() != 3)
+        return false;
+
+    // For (c) to be target of a Sliproad (ab) and (bc) have to be directed
+    auto backwards = intersection[0].entry_allowed;
+    auto undirected_link = intersection[1].entry_allowed && intersection[2].entry_allowed;
+
+    if (backwards || undirected_link)
+        return false;
+
+    return true;
+}
+
+double SliproadHandler::scaledThresholdByRoadClass(const double max_threshold,
+                                                   const RoadClassification &classification)
+{
+    double factor = 1.0;
+
+    switch (classification.GetPriority())
+    {
+    case RoadPriorityClass::MOTORWAY:
+        factor = 1.0;
+        break;
+    case RoadPriorityClass::TRUNK:
+        factor = 0.8;
+        break;
+    case RoadPriorityClass::PRIMARY:
+        factor = 0.8;
+        break;
+    case RoadPriorityClass::SECONDARY:
+        factor = 0.6;
+        break;
+    case RoadPriorityClass::TERTIARY:
+        factor = 0.5;
+        break;
+    case RoadPriorityClass::MAIN_RESIDENTIAL:
+        factor = 0.4;
+        break;
+    case RoadPriorityClass::SIDE_RESIDENTIAL:
+        factor = 0.3;
+        break;
+    case RoadPriorityClass::LINK_ROAD:
+        factor = 0.3;
+        break;
+    case RoadPriorityClass::CONNECTIVITY:
+        factor = 0.1;
```

Comment:
are these in relation to the differences we usually assume in speed? So do all of them relate to `speed_class/speed_motorway`?